core/Tracker/GoalManager.php

```diff
@@ -742,6 +742,7 @@ protected function insertNewConversion($conversion, $visitInformation, Request $
         Common::printDebug($newGoalDebug);
 
         $wasInserted = $this->getModel()->createConversion($conversion);
+        // TODO: if idsite/idorder are taken, but idvisit is different, this will silently fail. can we issue an error if possible?
```

Comment:
Issuing an error sounds good and useful to me 👍 Otherwise you have no chance to realise it failed.

Reply:
Throws an exception now if not inserted, though the error message doesn't appear in the tracker output (I think that's as it is normally).